API.md

```diff
@@ -0,0 +1,269 @@
+# API
+
+* [`create`](#create)
+* [`start`](#start)
+* [`stop`](#stop)
+* [`dial`](#dial)
+* [`dialProtocol`](#dialProtocol)
+* [`hangUp`](#hangUp)
+* [`handle`](#handle)
+* [`unhandle`](#unhandle)
+* [`peerRouting.findPeer`](#peerRouting.findPeer)
+* [`contentRouting.findProviders`](#contentRouting.findProviders)
+* [`contentRouting.provide`](#contentRouting.provide)
+* [`dht.put`](#dht.put)
+* [`dht.get`](#dht.get)
+* [`dht.findPeer`](#dht.findPeer)
+* [`dht.provide`](#dht.provide)
+* [`dht.findProviders`](#dht.findProviders)
+* [`dht.getClosestPeers`](#dht.getClosestPeers)
+* [`dht.getPublicKey`](#dht.getPublicKey)
+* [`pubsub.getPeersSubscribed`](#pubsub.getPeersSubscribed)
+* [`pubsub.getTopics`](#pubsub.getTopics)
+* [`pubsub.publish`](#pubsub.publish)
+* [`pubsub.subscribe`](#pubsub.subscribe)
+
+## create
+
+Creates an instance of Libp2p.
+
+### `create(options)`
+
+#### Parameters
+
+| Name | Type | Description |
+|------|------|-------------|
+| options | `Object` | libp2p options |
+
+#### Returns
+
+| Type | Description |
+|------|-------------|
+| `Promise<Libp2p>` | Promise resolves with the Libp2p instance |
+
+#### Example
+
+```js
+const { create } = require('libp2p')
+
+// specify options
+const options = {}
+
+// create libp2p
+const libp2p = await create(options)
+```
+
+Note: It is important pointing out that with `create`, the `PeerInfo` option is not required and will be generated if it is not provided.
+
+As an alternative, it is possible to create a Libp2p instance with the constructor:
+
+#### Example
+
+```js
+const Libp2p = require('libp2p')
+
+// specify options
+const options = {}
+
+// create libp2p
+const libp2p = new Libp2p (options)
+```
```

Comment:
js-ipfs had multiple rounds of iteration till it got its .create constructor https://github.com/ipfs/js-ipfs#ipfs-constructor, @alanshaw can shine some light here on why that was important. I would vote to have the same way of spawning a node just for consistency.

Reply:
I agree with the `create`. In `js-libp2p` we currently have `createLibp2p` and are moving into `create` now, same way as IPFS does.

I think this new API is consistent with `js-ipfs`. Same way, we also support using the `constructor`